add-extension.js

```diff
@@ -20,131 +20,150 @@ const { DH_UNABLE_TO_CHECK_GENERATOR } = require('constants');
 const readFile = util.promisify(fs.readFile);
 const writeFile = util.promisify(fs.writeFile);
 
-(async () => {
-  /**
-   * @type {{
-   *    extensions: {
-   *        id: string,
-   *        repository: string,
-   *        version?: string,
-   *        checkout?: string,
-   *        location?: string,
-   *        prepublish?: string,
-   *        extensionFile?: string,
-   *        download?: string
-   *    }[]
-   * }}
-   */
-  const { extensions } = JSON.parse(await readFile('./extensions.json', 'utf-8'));
-  const registry = new ovsx.Registry();
+const registry = new ovsx.Registry();
 
-  const argv = minimist(process.argv.slice(2));
-  if (argv._.length !== (!!argv.download ? 0 : 1)) {
-    console.log(`Usage: node add-extension REPOSITORY [OPTIONS]
-OPTIONS:
-    --checkout=CHECKOUT
-    --location=LOCATION
-    --prepublish=PREPUBLISH
-    --extensionFile=EXTENSION_FILE
-
-Alternative usage: node add-extension --download=VSIX_URL`);
-    process.exitCode = 1;
-    process.exit();
-  }
-
-  // Handle 'node add-extension --download=VSIX_URL':
-  if (argv.download) {
+// don't run this part if this module is not exported 
+// @ts-ignore
+if (!module.parent) {
+  (async () => {
+    const extensionFilePath = './extensions.json'
+    /**
+     * @type {{
+     *    extensions: {
+     *        id: string,
+     *        repository: string,
+     *        version?: string,
+     *        checkout?: string,
+     *        location?: string,
+     *        prepublish?: string,
+     *        extensionFile?: string,
+     *        download?: string
+     *    }[]
+     * }}
+     */
+    const { extensions } = await readExtensionsFromFile(extensionFilePath)
+  
+    const argv = minimist(process.argv.slice(2));
+    if (argv._.length !== (!!argv.download ? 0 : 1)) {
+      console.log(`Usage: node add-extension REPOSITORY [OPTIONS]
+  OPTIONS:
+      --checkout=CHECKOUT
+      --location=LOCATION
+      --prepublish=PREPUBLISH
+      --extensionFile=EXTENSION_FILE
+  
+  Alternative usage: node add-extension --download=VSIX_URL`);
+      process.exitCode = 1;
+      process.exit();
+    }
+  
+    // Handle 'node add-extension --download=VSIX_URL':
+    if (argv.download) {
+      try {
+        await exec('mkdir -p /tmp/vsix');
+        await exec(`wget -O extension.vsix ${argv.download}`, { cwd: '/tmp/vsix' });
+        await exec('unzip -q extension.vsix', { cwd: '/tmp/vsix' });
+        /** @type {{ publisher: string, name: string, version: string }} */
```

Comment:
The `ovsx` library should include type information, so this type comment can be removed.

Reply:
@spoenemann fixed